### ISSUE TITLE
[ENHANCE] Added Requirement to Level Up the TARDIS

### ISSUE THREAD

**Issue by Qanthelas** [DarkholmeTenk/TardisModIssues#299]
tl;dr: Add a config option to require something more than just TARDIS XP to level up the TARDIS.

I am thinking of making a modpack that would start the player out with a TARDIS but I don't want them to be able to quickly level it up and start traveling to any dimension they want.  I want to set a high energy level requirement to go to some dimensions to make them more "late game", but there's nothing to stop someone from quickly leveling the TARDIS and going there long before I want them to.

Thus my idea is to have the option to add some additional requirement to leveling up the TARDIS.  I am open to suggestions on what could be easy/feasible.  Some ideas that come to mind are:

1) An item that is consumed to level up the TARDIS, without this item the XP bar would fill but then max out until the item is supplied.  This could be either an item that must be put into the TARDIS console somehow or an item that must be in the player's inventory and a button could be pressed on the console to use it up to level up.  The recipe for this item could then be tweaked to keep players from leveling up their TARDIS (and thus preventing them from traveling to additional dimensions) until they acquire the items necessary to craft this upgrade item.

2) Require a visit to a certain dimension in order to pass a set level, all set as config options.  For example, the player could start with a Level 1 TARDIS, level it up to 2, but then couldn't level it up to 3 until they visit the Nether and then they could level it a few more times but couldn't pass the next cap until they visit the End, etc.  By setting the energy requirement for a given dimension to a level that couldn't be reached by upgrades prior to the level that unlocks by reaching that dimension, it would force the player to travel to a dimension for the first time in whatever "normal" way a player can travel rather than with the TARDIS (like a Nether Portal.)

3) An item that must be in the player's hand when they right click on something on the TARDIS console to level up.  This would require multiple items for different level caps, like a level 3 item, a level 5 item, etc. and again this would ideally be a config option.  This way, for example, I could limit a player from reaching level 3 until they have an Ender Pearl or from reaching level 5 until they have a Nether Star, or that kind of thing.

I am not sure which (if any) of these ideas are feasible, but I wanted to at least give some ideas.  The goal, though, is what I am after - to be able to limit a player from rapidly leveling up their TARDIS before completing objectives that I want them to complete.
